src/Avalonia.Base/Styling/EventSetter.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.PropertyStore;
+using Avalonia.Threading;
+
+namespace Avalonia.Styling;
+
+/// <summary>
+/// Represents an event setter in a style. Event setters invoke the specified event handlers in response to events.
+/// </summary>
+public sealed class EventSetter : SetterBase, ISetterInstance
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="EventSetter"/> class.
+    /// </summary>
+    public EventSetter()
+    {
+
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="EventSetter"/> class, using the provided event and handler parameters.
+    /// </summary>
+    /// <param name="eventName">The particular routed event that the <see cref="EventSetter"/> responds to.</param>
+    /// <param name="handler">The handler to assign in this setter.</param>
+    public EventSetter(RoutedEvent eventName, Delegate handler)
```

Comment:
I'd prefer something typed here rather than a plain Delegate. It kinda affects perf and had problems with NAOT previously (probably reflectionless one, don't remember)

Reply:
Shouldn't matter here.
Compiler still generates a typed `EventHandler<RoutedEventArgs>` instance. Which should give enough information to NAOT compiler.

It's then downcasted to Delegate, but our routed events infra uses Delegate internally anyway.

Reply:
There were problems with _invoking_ the delegate, IIRC

Reply:
Well, I sure can make EventSetter a generic type. And wire it up with some compiler magic.

Reply:
But I probably won't spend much time on it, unless @grokys approves these ISetterInstance hacks.